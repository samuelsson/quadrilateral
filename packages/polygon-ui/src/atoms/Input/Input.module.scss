@import '../../styles/abstracts/colors';

[data-theme='light'] {
  --Input-background-color: #{$color-white};
  --Input-border-color: #{$color-blue-gray-200};
  --Input-box-shadow: 0 4px 15px #{$color-gray-300};
}

[data-theme='dark'] {
  --Input-background-color: #{$color-blue-gray-700};
  --Input-border-color: #{$color-blue-gray-900};
  --Input-box-shadow: none;
}

.Input {
  height: 48px;
  padding: 0 16px;
  border-radius: 8px;
  border: 2px solid;
  border-color: var(--Input-border-color);
  box-sizing: border-box;
  background-color: var(--Input-background-color);
  box-shadow: var(--Input-box-shadow);
  color: inherit;
  font-size: 1rem;
  transition-property: box-shadow, border, opacity;
  transition-timing-function: linear;
  transition-duration: 0.2s;

  &:focus:not(:disabled),
  &:active:not(:disabled) {
    border-color: $color-blue;
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
}

.Error {
  border-color: $color-red;
}
