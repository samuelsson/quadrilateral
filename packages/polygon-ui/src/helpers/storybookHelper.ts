type HiddenArgTypes = {
  [key: string]: { table: { disable: boolean } };
};

// Used for hiding auto generated controls in stories.
export const hideControls = (args: string[]): HiddenArgTypes => {
  return args.reduce((result, item) => {
    return {
      ...result,
      [item]: { table: { disable: true } },
    };
  }, {});
};
