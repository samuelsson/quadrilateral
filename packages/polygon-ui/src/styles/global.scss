@import './abstracts/colors';

[data-theme='light'] {
  --global-background-color: #{$color-white};
  --global-color: #{$color-blue-gray-800};
}

[data-theme='dark'] {
  --global-background-color: #{$color-blue-gray-800};
  --global-color: #{$color-blue-gray-100};
}

body {
  background-color: var(--global-background-color);
  color: var(--global-color);
  font-family: sans-serif;
}
