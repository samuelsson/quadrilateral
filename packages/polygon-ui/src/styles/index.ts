export { colors } from './abstracts';
export { variables } from './abstracts';

export { default as global } from './global';
