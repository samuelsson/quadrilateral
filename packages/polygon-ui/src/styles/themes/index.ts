export { default as light } from './light';
export { default as dark } from './dark';

export { default as Theme } from './types';
