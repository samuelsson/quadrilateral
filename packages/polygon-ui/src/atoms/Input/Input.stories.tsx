import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, radios, text } from '@storybook/addon-knobs';

import Input from './Input';

storiesOf('Input', module).add(
  'Default',
  (): JSX.Element => (
    <Input
      type={radios('Type', { Text: 'text', Password: 'password' }, 'text')}
      name="input"
      placeholder={text('Placeholder', '')}
      disabled={boolean('Disabled', false)}
      error={boolean('Error', false)}
    />
  )
);
