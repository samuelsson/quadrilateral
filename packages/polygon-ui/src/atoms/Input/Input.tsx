import React from 'react';
import styled from 'styled-components';
import { colors } from '../../styles';
import { Theme } from '../../styles/themes';

interface InputProps {
  name: string;
  type: string;
  disabled?: boolean;
  error?: boolean;
  placeholder?: string;
  theme?: Theme;
}

const Input: React.FC<InputProps> = styled.input<InputProps>`
  height: 48px;
  padding: 0 16px;
  border-radius: 8px;
  border: 2px solid
    ${({ theme, error }): string =>
      error ? colors.red : theme.input.borderColor};
  box-sizing: border-box;
  background-color: ${({ theme }): string => theme.input.background};
  box-shadow: ${({ theme }): string => theme.input.boxShadow};
  color: inherit;
  font-size: 1rem;

  transition-property: box-shadow, border, opacity;
  transition-timing-function: linear;
  transition-duration: 0.2s;

  &:focus:not(:disabled),
  &:active:not(:disabled) {
    border-color: ${colors.blue};
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
`;

export default Input;
