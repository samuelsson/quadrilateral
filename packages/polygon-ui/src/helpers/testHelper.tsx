import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Theme } from '../styles/themes';

// For components using Themes and thus needing a `ThemeProvider` wrapper.
export const renderWithTheme = (
  children: React.ReactElement,
  theme: Theme
): RenderResult =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
