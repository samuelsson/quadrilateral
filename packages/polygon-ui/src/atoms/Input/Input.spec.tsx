import React from 'react';
import { renderWithTheme } from '../../helpers/testHelper';
import { light } from '../../styles/themes';
import Input from './Input';

describe('<Input />', (): void => {
  it('with light theme renders with no errors', (): void => {
    const { asFragment } = renderWithTheme(
      <Input name="input-light" type="text " />,
      light
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('<Input />', (): void => {
  it('with dark theme renders with no errors', (): void => {
    const { asFragment } = renderWithTheme(
      <Input name="input-dark" type="text " />,
      light
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
