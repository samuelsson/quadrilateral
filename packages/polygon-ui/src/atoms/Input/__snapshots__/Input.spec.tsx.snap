// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> with dark theme renders with no errors 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm ciemra"
    name="input-dark"
    type="text "
  />
</DocumentFragment>
`;

exports[`<Input /> with light theme renders with no errors 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm ciemra"
    name="input-light"
    type="text "
  />
</DocumentFragment>
`;
